docs/source/concepts/ledger-model/ledger-exceptions.rst

```diff
@@ -0,0 +1,288 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _da-model-exceptions:
+
+Exceptions
+----------
+
+The introduction of exceptions, a new Daml feature, has many implications
+for the ledger model. This page describes the changes to the ledger model
+introduced as part of this new feature.
+
+..
+   SF: Once the dust settles on exceptions, these changes should be
+   incorporated into the rest of the ledger model.
+
+Structure
++++++++++
+
+Under the new feature, Daml programs can raise and catch exceptions.
+When an exception is caught, the current subtransaction is rolled
+back to a certain point -- the beginning of the `try-catch` block.
+
+To support this in our ledger model, we need to modify the transaction
+structure to indicate which subtransactions were rolled back. We do this
+by introducing **rollback nodes** in the transaction. Each rollback nodes
+contain a rolled back subtransaction. Rollback nodes are not considered
+ledger actions, since there are too many differences between rollback
+nodes and .
+
+Therefore we define transactions as a list of **nodes**, where
+each node is either a ledger action or a rollback node. This is reflected
+in the updated EBNF grammar for the transaction structure:
+
+.. code-block:: none
+
+   Transaction  ::= Node*
+   Node         ::= Action | Rollback
+   Rollback     ::= 'Rollback' Transaction
+   Action       ::= 'Create' contract
+                  | 'Exercise' party* contract Kind Transaction
+                  | 'Fetch' party* contract
+                  | 'NoSuchKey' key
+   Kind         ::= 'Consuming' | 'NonConsuming'
+
+Note that `Action` and `Kind` have the same definition as before, but
+since `Transaction` may now contain rollback nodes, this means that an
+`Exercise` action may have a rollback node as one of its consequences.
+For example, ...
+
+..
+   TODO: Add example of an exercise containing a rollback node
+   with diagram. 'Alice orders her bank to "withdraw 1000 USD,
+   or, if it fails, to withdraw 500 USD instead".'
+
+In addition, rollback nodes may be nested. This corresponds to a possible
+case where multiple exceptions are raised and caught within the same
+transaction.
+
+Integrity
++++++++++
+
+The notion of a ledger action coming "after" another ledger action must
```

Comment:
I think only a few people will make the connection to the "before-after" relation from the consistency section. So I'd either give a direct pointer here. Even better would be to motivate the change by the example from the previous section, saying that intuitively this should be OK, but we need to adapt our integrity definition.

Reply:
Added a link.